src/compute-client/src/controller/instance.rs

```diff
@@ -608,26 +612,34 @@ impl<T: ComputeControllerTimestamp> Instance<T> {
         }
 
         let histogram_period = WallclockLagHistogramPeriod::from_epoch_millis(now_ms, &self.dyncfg);
+        let histogram_labels = match &self.workload_class {
+            Some(wc) => [("workload_class", wc.clone())].into(),
+            None => BTreeMap::new(),
+        };
 
         let mut histogram_updates = Vec::new();
+        let mut row_buf = Row::default();
         for (collection_id, collection) in &mut self.collections {
             if let Some(stash) = &mut collection.wallclock_lag_histogram_stash {
                 let lag = collection.shared.lock_write_frontier(|f| frontier_lag(f));
                 let bucket = lag.as_secs().next_power_of_two();
 
-                let key = (histogram_period, bucket);
+                let key = (histogram_period, bucket, histogram_labels.clone());
```

Comment:
This clone isn't nice, but I can't think of an easy way to avoid it.

Reply:
The whole handling of labels in `wallclock_lag_histogram_stash` is not great. Label values are fixed to strings right now, which will stop working once we add other types (like bools). Ideally we'd store a `Datum` but it's impossible to store an owned `String` as a `Datum`. I've been considering storing a `DatumMap` instead of the `BTreeMap`, but I don't know how to construct one from its contents. Should we store a `Row`?